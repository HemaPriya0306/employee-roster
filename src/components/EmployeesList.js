import React from 'react';
import './../assets/styles/employeesList.css';
import avatar from './../assets/images/account_circle.png';

const EmployeesList = ({ currentEmployees, onEmployeeClick, selectedEmp, searchValue }) => {
  return (
    <div className="emp-list-container" aria-live="polite">
      <table className="emp-table">
        <thead>
          <tr>
            <th>ID</th>
            <th>Avatar</th>
            <th>Name</th>
            <th>Contact Number</th>
            <th>Address</th>
          </tr>
        </thead>
        <tbody>
          {currentEmployees?.length > 0 ? (currentEmployees?.map((employee) => (
            <tr
              key={employee.id}
              className={selectedEmp?.id === employee.id ? 'selected-row' : ''}
              onClick={() => onEmployeeClick(employee)}
              tabIndex={0}
              onKeyDown={(e) => {
                if (e.key === 'Enter' || e.key === ' ') {
                  onEmployeeClick(employee);
                }
              }}
              role="button"
              aria-label={`View details for ${employee.firstName} ${employee.lastName}`}
            >
              <td>{employee.id}</td>
              <td className="emp-avatar-cell">
                <img alt={`${employee.firstName} ${employee.lastName}'s avatar`}
                  className="emp-avatar" src={avatar} />
              </td>
              <td>{employee.firstName} {employee.lastName}</td>
              <td>{employee.contactNo}</td>
              <td>{employee.address}</td>
            </tr>
          ))
          ) : (
            <tr>
              <td colSpan="5" style={{textAlign: "center"}}>No employees found</td>
            </tr>
          )}
        </tbody>
      </table>
    </div>
  );
};

export default EmployeesList;
