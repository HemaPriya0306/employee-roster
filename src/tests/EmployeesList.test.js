import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import EmployeesList from '../components/EmployeesList';

// Mock employee data for testing
const employees = [
  {
    id: 1,
    firstName: 'Test',
    lastName: 'User',
    contactNo: '1234567890',
    address: '123 xyz St',
  },
  {
    id: 2,
    firstName: 'Test',
    lastName: 'UserTwo',
    contactNo: '9876543210',
    address: '456 abc St',
  },
];

// Mock function for employee click
const mockEmployeeClick = jest.fn();

describe('EmployeesList component', () => {
  it('renders with no employees', () => {
    const { getByText } = render(
      <EmployeesList employees={[]} onEmployeeClick={mockEmployeeClick} selectedEmp={null} searchValue="" />
    );
    const noEmployeesMessage = getByText('No employees found.');
    expect(noEmployeesMessage).toBeInTheDocument();
  });

  it('renders with employees and matches snapshot', () => {
    const { container } = render(
      <EmployeesList employees={employees} onEmployeeClick={mockEmployeeClick} selectedEmp={null} searchValue="" />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders filtered employees based on searchValue', () => {
    const { getByText, queryByText } = render(
      <EmployeesList employees={employees} onEmployeeClick={mockEmployeeClick} selectedEmp={null} searchValue="Two" />
    );
    expect(getByText('Test UserTwo')).toBeInTheDocument();
    expect(queryByText('Test User')).not.toBeInTheDocument();
  });

  it('calls onEmployeeClick when an employee row is clicked', () => {
    const { getByText } = render(
      <EmployeesList employees={employees} onEmployeeClick={mockEmployeeClick} selectedEmp={null} searchValue="" />
    );
    const johnDoeRow = getByText('Test User');
    fireEvent.click(johnDoeRow);
    expect(mockEmployeeClick).toHaveBeenCalledWith(employees[0]);
  });

  it('calls onEmployeeClick when Enter key is pressed on an employee row', () => {
    const { getByText } = render(
      <EmployeesList employees={employees} onEmployeeClick={mockEmployeeClick} selectedEmp={null} searchValue="" />
    );
    const johnDoeRow = getByText('Test User');
    fireEvent.keyDown(johnDoeRow, { key: 'Enter', code: 'Enter' });
    expect(mockEmployeeClick).toHaveBeenCalledWith(employees[0]);
  });

  it('calls onEmployeeClick when Space key is pressed on an employee row', () => {
    const { getByText } = render(
      <EmployeesList employees={employees} onEmployeeClick={mockEmployeeClick} selectedEmp={null} searchValue="" />
    );
    const johnDoeRow = getByText('Test User');
    fireEvent.keyDown(johnDoeRow, { key: ' ', code: 'Space' });
    expect(mockEmployeeClick).toHaveBeenCalledWith(employees[0]);
  });
});
