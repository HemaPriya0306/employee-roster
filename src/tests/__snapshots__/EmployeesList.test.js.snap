// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmployeesList component renders with employees and matches snapshot 1`] = `
<div>
  <div
    aria-live="polite"
    class="emp-list-container"
  >
    <table
      class="emp-table"
    >
      <thead>
        <tr>
          <th>
            ID
          </th>
          <th>
            Avatar
          </th>
          <th>
            Name
          </th>
          <th>
            Contact Number
          </th>
          <th>
            Address
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          aria-label="View details for Test User"
          class=""
          role="button"
          tabindex="0"
        >
          <td>
            1
          </td>
          <td
            class="emp-avatar-cell"
          >
            <img
              alt="Test User's avatar"
              class="emp-avatar"
              src="account_circle.png"
            />
          </td>
          <td>
            Test
             
            User
          </td>
          <td>
            1234567890
          </td>
          <td>
            123 xyz St
          </td>
        </tr>
        <tr
          aria-label="View details for Test UserTwo"
          class=""
          role="button"
          tabindex="0"
        >
          <td>
            2
          </td>
          <td
            class="emp-avatar-cell"
          >
            <img
              alt="Test UserTwo's avatar"
              class="emp-avatar"
              src="account_circle.png"
            />
          </td>
          <td>
            Test
             
            UserTwo
          </td>
          <td>
            9876543210
          </td>
          <td>
            456 abc St
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
